src/Kernel/ClassOrganization.class.st

```diff
@@ -91,7 +91,7 @@ ClassOrganization >> classify: selector under: protocol [
 	oldProtocols := self protocolsOfSelector: selector.
 
 	self classify: selector inProtocolNamed: protocolName.
-	(oldProtocols select: #canBeRemoved) do: [ :e | self removeProtocol: e ].
+	(oldProtocols select: #canBeRemoved) do: [ :e | self removeProtocolIfEmpty: e ].
```

Comment:
`#canBeRemoved` → `#isEmpty` ? and if `removeProtocolIfEmpty` is worth its name, maybe the select is superfluous.

Reply:
Done thank you for the catch